File: scratch/check_ws.py
import asyncio
import websockets
import json
import sys

async def check_snapshot():
    uri = "ws://127.0.0.1:9001"
    try:
        async with websockets.connect(uri) as websocket:
            print(f"Connected to {uri}")
            # The first message should be the snapshot
            message = await websocket.recv()
            data = json.loads(message)
            
            if data.get("type") == "snapshot":
                print("Received Snapshot:")
                # Check for history in sessions
                sessions = data.get("sessions", [])
                print(f"Found {len(sessions)} sessions.")
                for i, s in enumerate(sessions):
                    history = s.get("history", [])
                    print(f"Session {i}: ID={s.get('acp_session_id')}, History Length={len(history)}")
                    if history:
                        print(f"  First event type: {history[0].get('type')}")
                
                # Output the whole snapshot for inspection if small, otherwise just summary
                # print(json.dumps(data, indent=2))
            else:
                print(f"Received unexpected event type: {data.get('type')}")
                print(json.dumps(data, indent=2))
                
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    asyncio.run(check_snapshot())

File: scratch/check_ws.js
const WebSocket = require('ws');

const ws = new WebSocket('ws://127.0.0.1:9001');

ws.on('open', function open() {
  console.log('Connected to ws://127.0.0.1:9001');
});

ws.on('message', function incoming(data) {
  const event = JSON.parse(data);
  if (event.type === 'snapshot') {
    console.log('Received Snapshot:');
    console.log(`Sessions count: ${event.sessions.length}`);
    event.sessions.forEach((s, i) => {
      console.log(`Session ${i}: ID=${s.acp_session_id}, History Length=${s.history ? s.history.length : 'undefined'}`);
      if (s.history && s.history.length > 0) {
          console.log(`  First event: ${JSON.stringify(s.history[0]).substring(0, 100)}...`);
      }
    });
    process.exit(0);
  } else {
    console.log('Received event:', event.type);
  }
});

ws.on('error', function error(err) {
  console.error('Error:', err);
  process.exit(1);
});

setTimeout(() => {
  console.log('Timeout waiting for snapshot');
  process.exit(1);
}, 5000);
